src/main/java/com/google/research/bleth/servlets/EnqueueSimulationServlet.java

```diff
@@ -0,0 +1,60 @@
+package com.google.research.bleth.servlets;
+
+import com.google.cloud.tasks.v2.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2.CloudTasksClient;
+import com.google.cloud.tasks.v2.HttpMethod;
+import com.google.cloud.tasks.v2.QueueName;
+import com.google.cloud.tasks.v2.Task;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet used for enqueuing a task targeted at endpoint '/new-simulation',
+ * in order to create and run a new simulation.
+ */
+@WebServlet("/enqueue-simulation")
+public class EnqueueSimulationServlet extends HttpServlet {
+
+    static final String projectId = "bleth-2020";
+    static final String locationId = "europe-west1";
+    static final String queueId = "simulations-queue";
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        try (CloudTasksClient client = CloudTasksClient.create()) {
+            // Construct the fully qualified queue name.
+            String queueName = QueueName.of(projectId, locationId, queueId).toString();
+
+            // Construct the HTTP request (for the task).
+            AppEngineHttpRequest httpRequest = AppEngineHttpRequest.newBuilder()
+                    .setRelativeUri("/new-simulation?" + toQueryString(request))
+                    .setHttpMethod(HttpMethod.POST)
+                    .build();
+
+            // Construct the task body.
+            Task task = Task.newBuilder()
+                    .setAppEngineHttpRequest(httpRequest)
+                    .build();
+
+            // Add the task to the queue.
+            client.createTask(queueName, task);
+        }
+
+        response.setContentType("text/html;");
+        response.getWriter().println("Task has been added to queue.");
+    }
+
+    private String toQueryString(HttpServletRequest request) {
+        List<String> keyValuePairs = new ArrayList<>();
+        for (String param : request.getParameterMap().keySet()) {
+            keyValuePairs.add(param + "=" + request.getParameter(param));
```

Comment:
Parameter values in a query string need to be URL-escaped.

Reply:
I thought of using [URLEncoder](https://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html#encode-java.lang.String-java.lang.String-):
```java
keyValuePairs.add(URLEncoder.encode(param) + "=" + URLEncoder.encode(request.getParameter(param)));
```
Is that what you mean?

Reply:
Using `URLEncoder` is indeed the right way to do it.

Reply:
Yes, using `UrlEncoder` this way is the right way to do it.